integrationtests/tests/tests.rs

```diff
@@ -1218,6 +1220,41 @@ async fn rejoin_consensus_threshold_peers() -> Result<()> {
     .await
 }
 
+#[tokio::test(flavor = "multi_thread")]
+async fn ecash_backup_can_recovery_metadata() -> Result<()> {
+    non_lightning_test(2, |_fed, user_send, _bitcoin, _, _| async move {
+        #[derive(Serialize, Deserialize)]
+        struct OurMetadata {
+            name: String,
+        }
+
+        let metadata = Metadata::json_from(OurMetadata {
+            name: "our_name".into(),
+        });
+
+        user_send
+            .client
+            .mint_client()
+            .back_up_ecash_to_federation(metadata.clone())
+            .await
+            .unwrap();
+
+        let mut task_group = TaskGroup::new();
+
+        assert_eq!(
+            user_send
+                .client
+                .mint_client()
+                .restore_ecash_from_federation(10, &mut task_group)
+                .await
+                .unwrap()
+                .unwrap(),
+            metadata
```

Comment:
hm, I found that a bit confusing that `restore_ecash_from_federation` returns the metadata now. Previously, it made sense to have `ecash` in the name as is recovered the ecash internally in the function but just return `()` (right?). Now, that the function is returning the metadata, perhaps rename it to `restore_backup_from_federation`? Internally, the functions continues to restore the ecash and it returns the metadata.

Reply:
The whole thing is a bit wonky now and will need to be replaced with a general purpose backup module, so I'll just leave as is for now.